code/game/objects/structures/gas_stand.dm

```diff
@@ -0,0 +1,240 @@
+
+	var/is_loosen = TRUE
+	var/valve_opened = FALSE
+
+/obj/structure/gas_stand/New()
+	..()
+	if (spawn_type)
+		tank = new spawn_type (src)
+	contained = new mask_type (src)
+	update_icon()
+
+/obj/structure/gas_stand/update_icon()
+	if (breather)
+		icon_state = "gas_stand_inuse"
+	else
+		icon_state = "gas_stand_idle"
+
+	overlays.Cut()
+
+	if (tank)
+		if(istype(tank,/obj/item/weapon/tank/anesthetic))
+			overlays += "tank_anest"
+		else if(istype(tank,/obj/item/weapon/tank/nitrogen))
+			overlays += "tank_nitro"
+		else if(istype(tank,/obj/item/weapon/tank/oxygen))
+			overlays += "tank_oxyg"
+		//else if(istype(tank,/obj/item/weapon/tank/phoron))
+		//	overlays += "tank_phoron"
+		//else if(istype(tank,/obj/item/weapon/tank/hydrogen))
+		//	overlays += "tank_hydro"
+		else
+			overlays += "tank_other"
+
+/obj/structure/gas_stand/Destroy()
+	STOP_PROCESSING(SSobj,src)
+	if(breather)
+		breather.internal = null
+		if(internalsHud)
+			internalsHud.icon_state = "internal0"
+	if(tank)
+		qdel(tank)
+	if(breather)
+		breather.remove_from_mob(contained)
+		src.visible_message("<span class='notice'>The mask rapidly retracts just before /the [src] is destroyed!</span>")
+	qdel(contained)
+	contained = null
+	breather = null
+	return ..()
+
+/obj/structure/gas_stand/attack_robot(var/mob/user)
+	if(Adjacent(user))
+		attack_hand(user)
+
+/obj/structure/gas_stand/MouseDrop(var/mob/living/carbon/human/target, src_location, over_location)
+	..()
+	if(istype(target))
+		if(!can_apply_to_target(target, usr)) // There is no point in attempting to apply a mask if it's impossible.
+			return
+		usr.visible_message("\The [usr] begins placing the mask onto [target]..")
+		if(!do_mob(usr, target, 25) || !can_apply_to_target(target, usr))
+			return
+		// place mask and add fingerprints
+		usr.visible_message("\The [usr] has placed \the mask on [target]'s mouth.")
+		attach_mask(target)
+		src.add_fingerprint(usr)
+		update_icon()
+		START_PROCESSING(SSobj,src)
+
+/obj/structure/gas_stand/attack_hand(mob/user as mob)
+	if (tank && is_loosen)
+		user.visible_message("<span class='notice'>\The [user] removes \the [tank] from \the [src].</span>", "<span class='notice'>You remove \the [tank] from \the [src].</span>")
+		user.put_in_hands(tank)
+		src.add_fingerprint(user)
+		tank.add_fingerprint(user)
+		tank = null
+		update_icon()
+		return
+	if (!tank)
+		to_chat(user, "<span class='warning'>There is no tank in \the [src]!</span>")
+		return
+	else
+		if (valve_opened)
+			src.visible_message("<span class='notice'>\The [user] closes valve on \the [src]!</span>")
+			if(breather)
+				if(internalsHud)
+					internalsHud.icon_state = "internal0"
+				breather.internal = null
+			valve_opened = FALSE
+			update_icon()
+		else
+			src.visible_message("<span class='notice'>\The [user] opens valve on \the [src]!</span>")
+			if(breather)
+				breather.internal = tank
+				if(internalsHud)
+					internalsHud.icon_state = "internal1"
```

Comment:
HUD icon should hadndled another way
Get hud element:
```
if(mob.HUDneed.Find("internal"))
	internalsHud = HUDneed["internal"]
```
Handle icon:
```
internalsHud.update_icon()
```

Reply:
>   LethalGhost approved these changes just now 
Oops! Missclicked.

Reply:
Im getting hud element at 119, im using variable here